libraries/Microsoft.Bot.Builder.Dialogs/DialogContainer.cs

```diff
@@ -32,6 +32,15 @@ protected DialogContainer(string dialogId = null)
         [JsonIgnore]
         public DialogSet Dialogs { get; set; } = new DialogSet();
 
+        /// <summary>
+        /// Gets or sets the referenced dialogs.
+        /// </summary>
+        /// <value>The dialogs referenced in this dialog contaner.</value>
+        [JsonProperty("dialogs")]
+#pragma warning disable CA2227 // Collection properties should be read only
+        public List<string> RefereceDialogs { get; set; } = new List<string>();
```

Comment:
>RefereceDialogs [](start = 28, length = 15)

Did you mean ReferencedDialogs?  I'm not sure I understand why this is necessary?

Reply:
Changed to ReferencedDialogs. 
Using ReferencedDialogs is suggested by Tom's idea.
My thought is if we load all .dialog files from resource loader. It is also not necessary to do EnsureDependenciesInstalled() in adaptive dialog. This ensures us only load the .dialog files we need.

Reply:
To @cosmicshuai, i think we don't need this extra property here, we should just use Dialogs property above. 

To @chrimc62, there is a no such a place for "all dialogs" and dialog manager doesn't hold that. This is the only place to register all the "possible referring dialogs". It's a little bit tricky, would love to chat about this with you.